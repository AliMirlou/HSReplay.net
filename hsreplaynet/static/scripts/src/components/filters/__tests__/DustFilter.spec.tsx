import React from "react";
import DustFilter from "../DustFilter";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("DustFilter", () => {
	test("renders correctly", () => {
		const component = renderer.create(
			<DustFilter dust={1600} setDust={jest.fn()} ownedDust={400} />,
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test("calls setDust if the form is submitted", () => {
		const setDust = jest.fn();
		const wrapper = mount(
			<DustFilter dust={1600} setDust={setDust} ownedDust={2000} />,
		);
		expect(wrapper).toMatchSnapshot("1600");
		expect(setDust).not.toHaveBeenCalled();
		(wrapper.find("input").instance() as any).value = "200";
		wrapper.find("input").simulate("change");
		wrapper.find("form").simulate("submit");
		expect(setDust).toHaveBeenCalledTimes(1);
		expect(setDust).toHaveBeenCalledWith(200);
		expect(wrapper).toMatchSnapshot("200");
	});

	test("calls setDust if the input is blurred", () => {
		const setDust = jest.fn();
		const wrapper = mount(
			<DustFilter dust={1600} setDust={setDust} ownedDust={2000} />,
		);
		expect(wrapper).toMatchSnapshot("1600");
		expect(setDust).not.toHaveBeenCalled();
		(wrapper.find("input").instance() as any).value = "200";
		wrapper.find("input").simulate("change");
		wrapper.find("input").simulate("blur");
		expect(setDust).toHaveBeenCalledTimes(1);
		expect(setDust).toHaveBeenCalledWith(200);
		expect(wrapper).toMatchSnapshot("200");
	});
});
