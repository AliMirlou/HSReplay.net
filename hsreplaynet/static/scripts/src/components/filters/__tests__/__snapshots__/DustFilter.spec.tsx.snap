// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DustFilter calls setDust if the form is submitted: 200 1`] = `
<DustFilter
  dust={1600}
  maxDust={50000}
  ownedDust={2000}
  setDust={
    [MockFunction] {
      "calls": Array [
        Array [
          200,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <form
    className="dust-filter"
    onSubmit={[Function]}
  >
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="common"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="common"
      >
        <Tooltip
          content="Spend 40 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-common.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="rare"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="rare"
      >
        <Tooltip
          content="Spend 100 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-rare.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="epic"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="epic"
      >
        <Tooltip
          content="Spend 400 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-epic.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="legendary"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="legendary"
      >
        <Tooltip
          content="Spend 1600 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset active"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-legendary.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="owned"
      value={2000}
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="owned"
        value={2000}
      >
        <Tooltip
          content="Spend all your dust (2000)"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/dust.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <input
      className="form-control"
      max={50000}
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      step={5}
      type="number"
      value="1600"
    />
  </form>
</DustFilter>
`;

exports[`DustFilter calls setDust if the form is submitted: 1600 1`] = `
<DustFilter
  dust={1600}
  maxDust={50000}
  ownedDust={2000}
  setDust={[MockFunction]}
>
  <form
    className="dust-filter"
    onSubmit={[Function]}
  >
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="common"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="common"
      >
        <Tooltip
          content="Spend 40 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-common.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="rare"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="rare"
      >
        <Tooltip
          content="Spend 100 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-rare.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="epic"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="epic"
      >
        <Tooltip
          content="Spend 400 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-epic.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="legendary"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="legendary"
      >
        <Tooltip
          content="Spend 1600 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset active"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-legendary.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="owned"
      value={2000}
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="owned"
        value={2000}
      >
        <Tooltip
          content="Spend all your dust (2000)"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/dust.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <input
      className="form-control"
      max={50000}
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      step={5}
      type="number"
      value="1600"
    />
  </form>
</DustFilter>
`;

exports[`DustFilter calls setDust if the input is blurred: 200 1`] = `
<DustFilter
  dust={1600}
  maxDust={50000}
  ownedDust={2000}
  setDust={
    [MockFunction] {
      "calls": Array [
        Array [
          200,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <form
    className="dust-filter"
    onSubmit={[Function]}
  >
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="common"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="common"
      >
        <Tooltip
          content="Spend 40 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-common.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="rare"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="rare"
      >
        <Tooltip
          content="Spend 100 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-rare.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="epic"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="epic"
      >
        <Tooltip
          content="Spend 400 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-epic.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="legendary"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="legendary"
      >
        <Tooltip
          content="Spend 1600 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset active"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-legendary.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="owned"
      value={2000}
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="owned"
        value={2000}
      >
        <Tooltip
          content="Spend all your dust (2000)"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/dust.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <input
      className="form-control"
      max={50000}
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      step={5}
      type="number"
      value="1600"
    />
  </form>
</DustFilter>
`;

exports[`DustFilter calls setDust if the input is blurred: 1600 1`] = `
<DustFilter
  dust={1600}
  maxDust={50000}
  ownedDust={2000}
  setDust={[MockFunction]}
>
  <form
    className="dust-filter"
    onSubmit={[Function]}
  >
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="common"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="common"
      >
        <Tooltip
          content="Spend 40 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-common.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="rare"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="rare"
      >
        <Tooltip
          content="Spend 100 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-rare.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="epic"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="epic"
      >
        <Tooltip
          content="Spend 400 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-epic.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="legendary"
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="legendary"
      >
        <Tooltip
          content="Spend 1600 dust"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset active"
              onClick={[Function]}
              src="/static/images/rarity-icons/rarity-legendary.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <Component
      isActive={[Function]}
      onClick={[Function]}
      type="owned"
      value={2000}
    >
      <DustPreset
        i18n={
          Object {
            "hasResourceBundle": [Function],
          }
        }
        isActive={[Function]}
        onClick={[Function]}
        t={[Function]}
        type="owned"
        value={2000}
      >
        <Tooltip
          content="Spend all your dust (2000)"
          noSrTooltip={true}
          simple={true}
        >
          <div
            className="tooltip-wrapper"
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onTouchStart={[Function]}
          >
            <img
              className="dust-preset"
              onClick={[Function]}
              src="/static/images/dust.png"
            />
          </div>
        </Tooltip>
      </DustPreset>
    </Component>
    <input
      className="form-control"
      max={50000}
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      step={5}
      type="number"
      value="1600"
    />
  </form>
</DustFilter>
`;

exports[`DustFilter renders correctly 1`] = `
<form
  className="dust-filter"
  onSubmit={[Function]}
>
  <div
    className="tooltip-wrapper"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchStart={[Function]}
  >
    <img
      className="dust-preset"
      onClick={[Function]}
      src="/static/images/rarity-icons/rarity-common.png"
    />
  </div>
  <div
    className="tooltip-wrapper"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchStart={[Function]}
  >
    <img
      className="dust-preset"
      onClick={[Function]}
      src="/static/images/rarity-icons/rarity-rare.png"
    />
  </div>
  <div
    className="tooltip-wrapper"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchStart={[Function]}
  >
    <img
      className="dust-preset"
      onClick={[Function]}
      src="/static/images/rarity-icons/rarity-epic.png"
    />
  </div>
  <div
    className="tooltip-wrapper"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchStart={[Function]}
  >
    <img
      className="dust-preset active"
      onClick={[Function]}
      src="/static/images/rarity-icons/rarity-legendary.png"
    />
  </div>
  <div
    className="tooltip-wrapper"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchStart={[Function]}
  >
    <img
      className="dust-preset"
      onClick={[Function]}
      src="/static/images/dust.png"
    />
  </div>
  <input
    className="form-control"
    max={50000}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    step={5}
    type="number"
    value="1600"
  />
</form>
`;
